contracts/DXswapFeeReceiver.sol

```diff
@@ -6,122 +6,226 @@ import './interfaces/IWETH.sol';
 import './libraries/TransferHelper.sol';
 import './libraries/SafeMath.sol';
 
-
 contract DXswapFeeReceiver {
-    using SafeMath for uint;
+    using SafeMath for uint256;
 
     address public owner;
     IDXswapFactory public factory;
     address public WETH;
```

Comment:
I think the framework has been to convert the fees, which start in the form of LP tokens, into the chain's native currency if possible.  This has made sense thus far with Ethereum Mainnet, Arbitrum (using ETH as native currency), and Gnosis Chain (using xDai).  However, I think generally speaking it's not obvious that the protocol fee should always be converted to native currency.  It could depend a lot on which chain.  Maybe we should consider making the target fee more flexible?  But in the context of how DXdao has been using the feeReceiver, then the above comment is correct that the intent is to be the chain's native currency and not always WETH.

Reply:
I think the approach of the second option makes sense.  The first implementation deployment of the feeReceiver was on mainnet and the recipient was the bonding curve, which needs to be sent ETH in order for it to work.  That fact probably influenced our initial implementation.  However, I don't think the new implementation needs to send to the bonding curve, as the bonding curve is now paused and in all likelihood will be sunset.

Reply:
I agree that the second option makes sense!
Ok, so if we decide not to swap to ETH at all then the `_swapTokensForETH` function can be removed as well as `withdraw` + `safeTransferETH`. Instead the new argument will be added like `targetToken`. By default feeReceiver will try to swap to `targetToken` which is ERC20 and if it's not possible send LP to the `fallbackReceiver`.
Am I right or misunderstood sth?